### ISSUE TITLE
Location being requested every minute (more often than actual weather requests)

### ISSUE THREAD

**Issue by alystair** [FelixdelasPozas/TrayWeather#174]
I just noticed with the latest Win 11 update that TrayWeather is checking my location every minute, even tho' I have it set to check for weather updates every 15 minutes.

Location checks should coincide with actual weather update requests, to reduce background use, and spamming the 'recent activity' log. 
![image](https://github.com/user-attachments/assets/a6026f3d-cb91-44a9-9685-e5ba2cb273f8)

**Comment by alystair** [comment]
Ok this is worse than I thought - even when selecting the 'manual coordinates' option in the 'Location' tab, Tray Weather continues to request location every minute.

**Comment by FelixdelasPozas** [comment]
Hi @alystair .

> Tray Weather continues to request location every minute.

I was going to ask you that precisely. I've got manual coordinates set and TrayWeather is not even in the list of activity. Do you have "roaming" checked?

I'll review the code for the next release, because I'm making many changes in that regard. Thanks for noticing and reporting it!

**Comment by alystair** [comment]
Nope, is there a config file somewhere? I could uninstall and start from scratch - maybe there is some ancient stuff that is triggering it.
![image](https://github.com/user-attachments/assets/755f5e38-6b69-4f5a-ae79-9533e36f214f)
![image](https://github.com/user-attachments/assets/1659d69d-90e1-4251-a08e-c2f62f847bef)

**Comment by FelixdelasPozas** [comment]
Thinking about adding a log tab into the configuration dialog with the latest accesses to network. Because the problem you're describing doesn't make sense to me (I'm pretty sure it's happening to you). You have the same config that I have, that is, fixed coords and no roaming.

One last question, do you have Win11 23H2 or 24H2? I have not yet updated to 24H2.

**Comment by alystair** [comment]
24H2 26100.1742

I'll do a clean reinstall - if that solves it it was most likely some old config remnants since I have been using this for some time :)

**Comment by FelixdelasPozas** [comment]
If you're not using the portable version (the one with the config in the INI file in the same directory of the application) the configuration is in the registry. In the key: HKEY_CURRENT_USER\Software\Felix de las Pozas Alvarez\TrayWeather

I doubt there is something strange there. In both you will find a key named "Use ip-api.com services" that will be false. I don't know why Windows thinks is requesting geo-location services every minute.

Anyway all the network requests code have been reworked for the next release (in a week or so I will release it). 

Regards, 
Félix.

**Comment by alystair** [comment]
Yep - nuked registry key between installs, might be nice to provide an option to erase that when uninstalling. Still appears in the recent activity list so this must be something on the MS side of things. Will await your next release 👍

**Comment by FelixdelasPozas** [comment]
Hi @alystair 

Probably today I'll release the next version, I'm still testing some things. I've added a way to see the network requests of the application, but it's not enabled by default. To log the requests and see them you'll have to start the application with the "--log" parameter. Then a new tab will appear in the Configuration dialog. 

![log](https://github.com/user-attachments/assets/305331d4-b6ee-4534-8415-ee08cebb672d)

From that tab you'll see the requests and the parameters. It's not updated in real time, but you can update with the button on the bottom of the tab. If the initial problem of the issue doesn't happen anymore let me know to close the issue.

Regards, 
Félix

**Comment by alystair** [comment]
I've installed 1.31.1 and so far I don't see any location requests appear even when reverting to IP Geolocation and enabling roaming - weird. Since this is low priority I'll let the new version run for a few days and check again.

**Comment by alystair** [comment]
Zero notifications since the update, considering this fixed - you should add it to the change log. Thank you!
